misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ServiceDeclarationNodeVisitor.java

```diff
@@ -64,16 +65,17 @@
 public class ServiceDeclarationNodeVisitor extends NodeVisitor {
 
     private final SemanticModel semanticModel;
+    private final SyntaxTree syntaxTree;
     private final ComponentModel.PackageId packageId;
     private final Package currentPackage;
     private final List<Service> services = new LinkedList<>();
     private final Path filePath;
 
-    public ServiceDeclarationNodeVisitor(SemanticModel semanticModel, Package currentPackage,
+    public ServiceDeclarationNodeVisitor(SemanticModel semanticModel, SyntaxTree syntaxTree, Package currentPackage,
                                          ComponentModel.PackageId packageId, Path filePath) {
```

Comment:
do we need to pass both of the package and packageID as constructor arguments?  Can't we always derive the packageID from the package instance?